File: src/main/java/com/cfgglobal/test/dao/UserDao.java
package com.cfgglobal.test.dao;

import com.cfgglobal.test.dao.base.BaseDao;
import com.cfgglobal.test.domain.User;
import io.vavr.control.Option;
import org.springframework.stereotype.Repository;

@Repository
public interface UserDao extends BaseDao<User, Long> {

    Option<User> findByUsername(String name);

}

File: src/main/kotlin/com/cfgglobal/test/dao/UserDao.kt
package com.cfgglobal.test.dao

import com.cfgglobal.test.dao.base.BaseDao
import com.cfgglobal.test.domain.User
import org.springframework.stereotype.Repository

@Repository
interface UserDao : BaseDao<User, Long> {

    fun findByUsername(name: String): User

}